app/lib/admin/system_check/elasticsearch_check.rb

```diff
@@ -49,5 +77,21 @@ def compatible_version?
 
     Gem::Version.new(running_version) >= Gem::Version.new(required_version) ||
       Gem::Version.new(compatible_wire_version) >= Gem::Version.new(required_version)
+  rescue ArgumentError
+    false
+  end
+
+  def mismatched_indexes
+    @mismatched_indexes ||= INDEXES.filter_map do |klass|
+      klass.index_name if Chewy.client.indices.get_mapping[klass.index_name]&.deep_symbolize_keys != klass.mappings_hash
+    end
+  end
+
+  def indexes_match?
+    mismatched_indexes.empty?
+  end
+
+  def preset_matches?
+    (cluster_health['number_of_nodes'] < 2) == (ENV.fetch('ES_PRESET', 'single_node_cluster') == 'single_node_cluster')
```

Comment:
Could this be written more clearly? I find it hard to understand the exact conditions